pvlib/iotools/merra2.py

```diff
@@ -0,0 +1,154 @@
+import pandas as pd
+import requests
+from io import StringIO
+
+
+VARIABLE_MAP = {
+    'SWGDN': 'ghi',
+    'SWGDNCLR': 'ghi_clear',
+    'ALBEDO': 'albedo',
+    'T2M': 'temp_air',
+    'T2MDEW': 'temp_dew',
+    'PS': 'pressure',
+    'TOTEXTTAU': 'aod550',
+}
+
+
+def get_merra2(latitude, longitude, start, end, username, password, dataset,
+               variables, map_variables=True):
+    """
+    Retrieve MERRA-2 time-series irradiance and meteorological data from
+    NASA's GESDISC data archive.
+
+    MERRA-2 [1]_ offers modeled data for many atmospheric quantities at hourly
+    resolution on a 0.5° x 0.625° global grid.
+
+    Access must be granted to the GESDISC data archive before EarthData
+    credentials will work.  See [2]_ for instructions.
+
+    Parameters
+    ----------
+    latitude : float
+        In decimal degrees, north is positive (ISO 19115).
+    longitude: float
+        In decimal degrees, east is positive (ISO 19115).
+    start : datetime like or str
+        First timestamp of the requested period. If a timezone is not
+        specified, UTC is assumed.
+    end : datetime like or str
+        Last timestamp of the requested period. If a timezone is not
+        specified, UTC is assumed.
+    username : str
+        NASA EarthData username.
+    password : str
+        NASA EarthData password.
+    dataset : str
+        Dataset name (with version), e.g. "M2T1NXRAD.5.12.4".
+    variables : list of str
+        List of variable names to retrieve.  See the documentation of the
+        specific dataset you are accessing for options.
+    map_variables : bool, default True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+
+    Raises
+    ------
+    ValueError
+        If ``start`` and ``end`` are in different years, when converted to UTC.
+
+    Returns
+    -------
+    data : pd.DataFrame
+        Time series data. The index corresponds to the middle of the interval.
+    meta : dict
+        Metadata.
+
+    Notes
+    -----
+    The following datasets provide quantities useful for PV modeling:
+
+    - M2T1NXRAD.5.12.4: SWGDN, SWGDNCLR, ALBEDO
+    - M2T1NXSLV.5.12.4: T2M, U10M, V10M, T2MDEW, PS, TO3, TQV
+    - M2T1NXAER.5.12.4: TOTEXTTAU, TOTSCATAU, TOTANGSTR
+
```

Comment:
Would it be possible to include all these in the variable map?  And make that visible in the documentation?

Also, if there is down-welling long-wave, please add it for module temperature modeling.

Reply:
I think the remaining items do not have pvlib names.

Several long-wave variables are available (click to the "Variables" tab): https://disc.gsfc.nasa.gov/datasets/M2T1NXRAD_5.12.4/summary

I guess `LWGNT` may be the most relevant.  Perhaps someone more familiar with this area can confirm?

Reply:
> @AdamRJensen, what is the time resolution of the data in your plots, and/or does it include nighttime? I would expect some negative values at night from the BSRN measurements, at least on clear nights. And should it always be close to zero or negative if it's a longwave sensor shaded from beam irradiance?

I've added a few more scatter plots above. I don't think that downwelling longwave is ever greater than upwelling. This would mean that the effective sky temperature of the clouds would be higher than that of the surface: unlikely, but maybe not impossible.

Reply:
Added `'LWGNT': 'lwn', 'LWGEM': 'lwu'` in latest commit